### PR TITLE
Add epacamd_eia analysis tests

### PR DESCRIPTION
**Pull request by aesharpe**
I merged an old version of the crosswalk analysis module tests into `add-epacamd-eia-analysis-test` so that I could compare the new changes to the old version before merging it into dev as a whole new file.

### CODE REVIEW

File: test/unit/analysis/epacamd_eia_test.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    # fix column order and drop some columns
+    expected = expected[
+        ["subplant_id", "plant_id_eia", "emissions_unit_id_epa", "generator_id"]
+    ]
+
+    # should be two separate tests but I ran out of time
+    actual1 = cw_analysis.filter_crosswalk(mock_crosswalk, mock_cems_extended)
+    actual = cw_analysis.make_subplant_ids(actual1)
+    print(actual)
+    print(expected)
```
Comment (TrentonBush) [inline test/unit/analysis/epacamd_eia_test.py:197]: debugging prints accidentally left in?

File: test/unit/analysis/epacamd_eia_test.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+        generator_id=[0, 1, 1, 2, 0, 0, 0, 1],
+        subplant_id=[0, 1, 1, 1, 0, 0, 0, 0],
+    )
+    # fix column order and drop some columns
+    expected = expected[
+        ["subplant_id", "plant_id_eia", "emissions_unit_id_epa", "generator_id"]
+    ]
+
+    # should be two separate tests but I ran out of time
+    actual1 = cw_analysis.filter_crosswalk(mock_crosswalk, mock_cems_extended)
```
Comment (TrentonBush) [inline test/unit/analysis/epacamd_eia_test.py:194]: what is the corresponding `expected1`? I'd say either add that and test it or rename `actual1` so it doesn't give that false impression that it is being tested (at least that's what I think when I see the "actual" variable in a test)

File: test/unit/analysis/epacamd_eia_test.py [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    many_to_many = uniques.query('plant_id_eia == 10 and emissions_unit_id_epa == "c"')
+    expected = (
+        pd.concat([uniques, one_to_many, many_to_many])
+        .sort_index()
+        .reset_index(drop=True)
+    )
+
+    expected = expected.assign(
+        plant_id_eia=expected["plant_id_eia"],
+        emissions_unit_id_epa=expected["emissions_unit_id_epa"],
```
Comment (TrentonBush) [inline test/unit/analysis/epacamd_eia_test.py:141]: Do the first two assignments do anything?

File: test/unit/analysis/epacamd_eia_test.py [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
```
Comment (TrentonBush) [inline test/unit/analysis/epacamd_eia_test.py:184]: Same with these

**Review by TrentonBush — APPROVED** [review]
Reading this a year later, the pattern of "use a simple fixture and have each test bolt on additional complexity as needed" is kind of confusing. Instead, I wish the fixtures had the most complexity and had long docstrings explaining what each part was for. Then the tests could subset the complex thing if needed. But I probably didn't know that until the end 🤷🏼‍♂️

File: test/unit/analysis/epacamd_eia_test.py (hunk 1 above)
Comment (aesharpe) [reply test/unit/analysis/epacamd_eia_test.py:197]: yup, thanks for catching that.

File: test/unit/analysis/epacamd_eia_test.py (hunk 3 above)
Comment (aesharpe) [reply test/unit/analysis/epacamd_eia_test.py:141]: Ya, they pull in the plant and emissions ids from another table that was created. It's a little bit of a 'round-about, but I didn't feel like editing the previous version more than I had to.

File: test/unit/analysis/epacamd_eia_test.py (hunk 4 above)
Comment (aesharpe) [reply test/unit/analysis/epacamd_eia_test.py:184]: See above.

File: test/unit/analysis/epacamd_eia_test.py (hunk 3 above)
Comment (TrentonBush) [reply test/unit/analysis/epacamd_eia_test.py:139]: Isn't it just referencing itself to assign a column that already exists?

**Pull request merged**